src/main/java/core/basesyntax/strategy/impl/SupplyOperation.java

```diff
@@ -5,12 +5,16 @@
 import core.basesyntax.model.FruitTransaction;
 
 public class SupplyOperation implements OperationHandler {
-    private final FruitDao storageDao = new FruitDaoImpl();
+    private final FruitDao storageDao;
+
+    public SupplyOperation() {
+        this.storageDao = new FruitDaoImpl();
```

Comment:
Not yet)
Created only one fruitDao instance and pass to the constructor params everywhere